categories/views.py

```diff
@@ -1,10 +1,22 @@
 from django.http import HttpRequest, HttpResponse
 from django.shortcuts import render
 
-from categories.models import MainCategory
+from categories.models import Category, MainCategory
+from posts.models import Post
 
 
 def category_list(request: HttpRequest) -> HttpResponse:
     objects = MainCategory.objects.prefetch_related("category_set").all()
 
     return render(request, "categories/category_list.html", {"objects": objects})
+
+
+def category_detail(request: HttpRequest, category_id: int) -> HttpResponse:
+    objects = Post.objects.filter(category_id=category_id, is_active=1)
+    category = Category.objects.filter(pk=category_id)
```

Comment:
better always check id or return 404
```suggestion
    category = get_object_or_404(category_id)
    posts = Post.objects.filter(category_id=category.id, is_active=True)
```

Reply:
category = get_object_or_404(category_id)   how it understand what i need if model doesn't specified

Reply:
Done)